src/probability/independence.lean

```diff
@@ -342,4 +342,30 @@ lemma indep_sets.indep_set_of_mem (hs : s ∈ S) (ht : t ∈ T) (hs_meas : measu
 
 end indep_set
 
+section indep_fun
+
+variables {α β β' γ γ' : Type*} [measurable_space α] [measurable_space β] [measurable_space β']
+variables [measurable_space γ] [measurable_space γ'] {μ : measure α}
+
+lemma indep_fun_of_indep_fun_of_ae_eq {f g f' g' : α → β} (hfg : indep_fun f g μ)
+  (hf : f =ᵐ[μ] f') (hg : g =ᵐ[μ] g') : indep_fun f' g' μ :=
+begin
+  rintro _ _ ⟨A,hA,rfl⟩ ⟨B,hB,rfl⟩,
+  have h1 : f ⁻¹' A =ᵐ[μ] f' ⁻¹' A := hf.fun_comp A,
+  have h2 : g ⁻¹' B =ᵐ[μ] g' ⁻¹' B := hg.fun_comp B,
+  rw [←measure_congr h1, ←measure_congr h2, ←measure_congr (h1.inter h2)],
+  exact hfg _ _ ⟨_,hA,rfl⟩ ⟨_,hB,rfl⟩
+end
+
+lemma indep_fun_comp_of_indep_fun {f : α → β} {g : α → β'} (hfg : indep_fun f g μ)
+  {φ : β → γ} {ψ : β' → γ'} (hφ : measurable φ) (hψ : measurable ψ) :
+  indep_fun (φ ∘ f) (ψ ∘ g) μ :=
+begin
+  rintro _ _ ⟨A,hA,rfl⟩ ⟨B,hB,rfl⟩, apply hfg,
+  exact ⟨φ ⁻¹' A, hφ hA, set.preimage_comp.symm⟩,
+  exact ⟨ψ ⁻¹' B, hψ hB, set.preimage_comp.symm⟩
```

Comment:
Since there are now two sub-goals, they should be in `{ }`.
```suggestion
  { exact ⟨φ ⁻¹' A, hφ hA, set.preimage_comp.symm⟩ },
  { exact ⟨ψ ⁻¹' B, hψ hB, set.preimage_comp.symm⟩ }
```

Reply:
Sure, I always do that for subgoals whose proofs are not one-liners but it makes sense to be systematic :-)